proxy/rpc/np_gas_price.py

```diff
@@ -140,3 +168,122 @@ async def get_neon_gas_price(
+        if not priority_fee_percentiles:
+            # In case the User was lazy to request any percentiles, let's return "median"
+            # so the response makes sense.
+            priority_fee_percentiles = [50]
+        # Validate percentiles first before querying anything.
+        for p in priority_fee_percentiles:
+            if p < 0 or p > 100:
+                return None
+
+        # Fetch the latest block slot.
+        block = await self.get_block_by_tag(block_tag)
+        # Max value for the bigint type.
+        # In case block is latest and it's empty, let's use this max value
+        # instead of making an extra get_latest_block DB query.
+        latest_slot = 9223372036854775807
+        if not block.is_empty:
+            latest_slot = block.slot
+
+        # Let's clamp num_blocks to not create excessive load and attack vector from RPC clients.
+        # Infura and Alchemy currently do the same.
+        num_blocks = max(1, min(num_blocks, 1024))
+        # Fetching the data stage.
+        fee_gas_data: list[BlockFeeGasData] = await self._db.get_historical_base_fees(
+            self.get_chain_id(ctx), num_blocks, latest_slot
+        )
+        priority_fees_data: list[list[int]] = await self._db.get_historical_priority_fees(num_blocks)
+
+        # Since we are effectively querying different tables, let's reconcile length of the data,
+        # so we return consistent data (in case one array is bigger than the other).
+        num_blocks = min(len(fee_gas_data), len(priority_fees_data))
+        if num_blocks == 0:
+            # For some reason, the number of "blocks" we have is zero, so return the current base fee price.
+            _, token_gas_price = await self.get_token_gas_price(ctx)
+            current_gas_price: int = token_gas_price.suggested_gas_price
+            return _RpcFeeHistoryResp.from_raw([current_gas_price], [], 0, [])
+
+        fee_gas_data = fee_gas_data[-num_blocks:]
+        priority_fees_data = priority_fees_data[-num_blocks:]
+
+        # Processing stage. Data entries are sorted in descending order by slot.
+        # BaseFees:
+        base_fees: list[int] = [
+            math.ceil(fee_gas_data[idx].average_base_fee) for idx in range(len(fee_gas_data) - 1, -1, -1)
+        ]
+        # Since ethereum deterministically derives the next base_fee_per_gas for the upcoming block,
+        # we have to do the same - return the current gas price (the same as in the eth_gasPrice).
+        _, token_gas_price = await self.get_token_gas_price(ctx)
+        current_gas_price: int = token_gas_price.suggested_gas_price
+        base_fees.append(current_gas_price)
+
+        # GasUsedRatio:
+        gas_used_ratio = [
+            min(1.0, fee_gas_data[idx].total_gas_used / 48_000_000_000_000)
+            for idx in range(len(fee_gas_data) - 1, -1, -1)
+        ]
+
+        # OldestBlock:
+        oldest_block = fee_gas_data[-1].block_slot if len(fee_gas_data) > 0 else 0
```

Comment:
This simplifying is enough:
`if fee_gas_data_list else 0`

Reply:
done